addons/project_timesheet_holidays/tests/test_timesheet_global_time_off.py

```diff
@@ -121,6 +126,15 @@ def test_timesheet_creation_and_deletion_on_employee_archive(self):
             'date_to': leave_end_datetime,
         })
 
+        # Create a global time-off in another company which should not be
+        # taken into account when creating/unarchiving employee
+        self.env['resource.calendar.leaves'].with_company(self.test_company_2).create({
+            'name': 'New Year 2023',
+            'calendar_id': False,
+            'date_from': datetime(2023, 1, 1, 0, 0, 0),
+            'date_to': datetime(2023, 12, 31, 23, 59, 59),
+        })
```

Comment:
```suggestion
        self.env['resource.calendar.leaves'].with_company(self.test_company_2).create({
            'name': 'Global leave in another company',
            'calendar_id': False,
            'date_from': datetime(2022, 2, 1, 0, 0, 0),
            'date_to': datetime(2022, 2, 7, 23, 59, 59),
        })
```
It is to avoid generating timesheets for the whole year (2023) if one day we break the test. 🙂